gitops/snapshot.go

```diff
@@ -235,6 +237,17 @@ func IsSnapshotValid(snapshot *applicationapiv1alpha1.Snapshot) bool {
 	return false
 }
 
+// IsImageDigestValid checks if image url contains digest and it is valid image, return false if check fails
+func IsImageDigestValid(imageUrl string) bool {
+	ref, err := name.NewDigest(imageUrl)
+	if err != nil {
+		return false
+	}
+	_, err = remote.Get(ref)
```

Comment:
It looks like `remote.Get()` queries the registry, which is fine for public repositories, but would it cause issues when accessing private ones?

Reply:
Good point. I get this page https://github.com/google/go-containerregistry/blob/main/pkg/authn/README.md and let me update the code to use `img, err := remote.Get(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain))` and return `nil` if fails. I think it is supposed that `DOCKER_CONFIG` exist in the deployed environment.

Reply:
@dirgim If `DOCKER_CONFIG` or `${XDG_RUNTIME_DIR}` is not defined in the environment, should we require it to be provided in log, especially for the external users? Or we just verify the digest of containerImage is a vaid sha string?

Reply:
Updated.

Reply:
Thanks, looks good!